### PR TITLE
[IN-2039] BaseSimulators packer and playbook

### PR DESCRIPTION
**Pull request by bitrise-tom**
### New Pull Request Checklist

*Please mark the points which you did / accept.*

- [ ] The tool I added is stable, and __does not require frequent updates__. _Preinstalled tools on the LTS stacks
  are not updated, so if the tool requires frequent updates you should handle the installation on-demand (see: [Install Any Additional Tool - DevCenter](https://bitrise-io.github.io/devcenter/tips-and-tricks/install-additional-tools/) for more information)_
- [ ] I updated the corresponding tests if there were any.
- [ ] I added a version report line to [`system_report.sh`](/system_report.sh) for the new tool(s) I added
- [ ] I have added an entry to the CHANGELOG.md with a revisionID (the current date and the number of the change e.g.: 2019_10_11_1) and a brief description of the actual change
- [ ] I have updated the revisionID in [bitrise_profile](roles/profiles/files/bitrise_profile) **BITRISE_OSX_STACK_REV_ID** with the one I just created in the CHANGELOG.md

### CODE REVIEW

File: baseSimulators-playbook.yml [hunk 1]
```diff
@@ -0,0 +1,15 @@
+---
+#
+# Ansible playbook to prepare a Bitrise OS X VM/box, used through vagrant
+#
+
+- hosts: all
+  become_method: sudo
```
Comment (lmesz-bitrise) [inline baseSimulators-playbook.yml:7]: Shouldn't the user need to be added? Something like `remote_user: vagrant`

File: packer/baseSimulators.json [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+      "user": "vagrant",
+      "extra_arguments": [ "--extra-vars", "xcode_version={{user `xcode_version`}}" ],
+      "ansible_env_vars": [ "ANSIBLE_HOST_KEY_CHECKING=False", "ANSIBLE_SSH_ARGS='-o ForwardAgent=yes -o ControlMaster=auto -o ControlPersist=60s'", "ANSIBLE_NOCOLOR=True" ],
+      "playbook_file": "../baseSimulators-playbook.yml"
+    }
+  ],
+  "variables": {
+    "cpu_count": "4",
+    "disk_size": "300G",
+    "ram_size": "16G",
```
Comment (lmesz-bitrise) [inline packer/baseSimulators.json:34]: A beefy one. :)

File: packer/baseSimulators.json [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+    "xcode_version": "{{env `XCODE_VERSION`}}",
```
Comment (lmesz-bitrise) [inline packer/baseSimulators.json:35]: Doesn't need to do the variables -> variables trick?
I mean first the packer file should have a variables section and env vars should be used there and after that using the user `variable_name` thing like we do in vsphere-management-utils repo?

File: baseSimulators-playbook.yml (hunk 1 above)
Comment (bitrise-tom) [reply baseSimulators-playbook.yml:7]: no, packer is setup with vagrant as the default user and it is passed properly

File: packer/baseSimulators.json (hunk 3 above)
Comment (bitrise-tom) [reply packer/baseSimulators.json:35]: that is what we are doing here, only the variables are here at the end :)

File: baseSimulators-playbook.yml [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+  any_errors_fatal: true
+  vars:
+    - param_user: vagrant
+    - XCODE_INSTALL_USER: <XXXX>
+    - XCODE_INSTALL_PASSWORD: <XXXX>
```
Comment (fehersanyi-bitrise) [inline baseSimulators-playbook.yml:12]: where, or when do you plan to inject these? 🧐

File: packer/baseSimulators.json (hunk 3 above)
Comment (lmesz-bitrise) [reply packer/baseSimulators.json:35]: Doh, sorry :)

Comment (bitrise-tom) [reply packer/baseSimulators.json:35]: np

File: baseSimulators-playbook.yml (hunk 4 above)
Comment (bitrise-tom) [reply baseSimulators-playbook.yml:12]: when we move to the automation phase. Let's just get some images building first

**Review by lmesz-bitrise — APPROVED** [review]

**Review by Kanan-Bitrise — APPROVED** [review]

**Pull request merged**